app.go

```diff
@@ -193,9 +198,43 @@ func New(opts ...Option) *App {
 		app.provide(p)
 	}
 	app.provide(func() Lifecycle { return app.lifecycle })
+
+	if app.err != nil {
+		app.logger.Printf("Error after options were applied: %v", app.err)
+		return app
+	}
+
+	if err := app.executeInvokes(); err != nil {
+		app.err = err
```

Comment:
I like `app.Err` better than `TryNew`.

```
app := New(..)
app.Run()
```

Is equivalent to,

```
app := New(..)

if err := app.Err(); err != nil {
  panic(err)
}

// Usual Start/Stop logic
```

`Start` would need to account for `Err` not having been checked manually by the user.

(Also, let's call it `Err()` instead. There's precedent: https://golang.org/pkg/bufio/#Scanner.Err)